Sources/ReadingListKit/ReadingListModel.swift

```diff
@@ -0,0 +1,154 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Combine
+import MoSoAnalytics
+
+protocol ReadingListModelDelegate: AnyObject {
+    func didFetchReadingListItems(urlStrings: [String], totalItemCount: Int, cursor: String?)
+    func removeItemFromList(item: String)
+    func operationFailed(with error: PocketAccessLayerError)
+}
+
+enum ReadingListDisplayMode {
+    case normal
+    case empty
+    case error
+    case loggedOut
+}
+
+public class ReadingListModel: ReadingListModelDelegate, ObservableObject {
+    private let analytics: ReadingListTracker
+
+    var pocketAccessLayer: PocketAccessLayerProtocol
+    var totalNumberOfItemsInReadingList: Int?
+    @Published private(set) var displayMode: ReadingListDisplayMode = .normal
+    @Published private(set) var readingListItems: [ReadingListCellViewModel] = []
+    private var paginationCursor: String?
+
+    public init(sessionProvider: @escaping ReadingListSessionProvider, consumerKey: String, analyticsTracker: ReadingListTracker) {
+        self.analytics = analyticsTracker
+        pocketAccessLayer = PocketAccessLayer(sessionProvider, consumerKey)
+        pocketAccessLayer.delegate = self
+
+        fetchMoreReadingList()
+    }
+
+    // MARK: - Fetch Reading List Items
+
+    func didDisplay(item: ReadingListCellViewModel) {
```

Comment:
One thing that's throwing me off, here, is that we're using a view model as an identifier for an item, whereas in other places we're using a string (I believe its URL…?). The latter makes more sense to be, since the supplied view model here is a representation of what should be presented, and not necessarily an item itself. 

What do you think about updating this function to take in a `String` rather than a view model?

Reply:
I agree that this is a bit clunky, the reason I did it was that it was easier to filter the main array by the VM rather than reaching into each to find it's URL.
But you're right, that is a better way and only slightly more effort.

Reply:
Updated!